ssl/t1_lib.c

```diff
@@ -2363,7 +2398,8 @@ int tls_choose_sigalg(SSL *s, int fatalerrs)
                 return 1;
 
         if (SSL_USE_SIGALGS(s)) {
-            if (s->s3->tmp.peer_sigalgs != NULL) {
+            if (s->s3->tmp.peer_cert_sigalgs != NULL
```

Comment:
I'm slightly confused by this change. How does the value of `peer_cert_sigalgs` ever get used in the TLSv1.2 case?

Reply:
In the present form of this pull request, the value does not get used in this block.  (I may have lost track that this is in the else clause of a TLS13 block, when writing it.)  The motivation was along the lines of that if the peer is making a claim about what signature algorithms to use, then we should not fall back to a default list if there is no overlap in the provided values.  That said, looking back at the change with fresh eyes, the practical effect is to forbid the case of sending signature_algorithms_cert and not signature_algorithms.  The spec does say "TLS 1.2 implementations SHOULD also process this extension [signature_algorithms_cert].", so it does seem spec-compliant for a client to only offer TLS 1.2, send signature_algorithms_cert, but request the default sigalgs by not sending the regular signature_algorithms.  That implies that this check should be removed, but the block to pick a cert should be copied from the TLS13 case and duplicated here.  (Given the conditions for when the default sigalgs should be used, I'm not sure that it's a win to try to consolidate the signature_algorithms_cert block.)